core/src/banking_stage/transaction_scheduler/transaction_state.rs

```diff
@@ -33,18 +31,18 @@ pub(crate) struct SanitizedTransactionTTL {
 ///   to the appropriate thread for processing. This is done to avoid cloning the
 ///  `SanitizedTransaction`.
 #[allow(clippy::large_enum_variant)]
-pub(crate) enum TransactionState {
+pub(crate) enum TransactionState<Tx> {
     /// The transaction is available for scheduling.
     Unprocessed {
-        transaction_ttl: SanitizedTransactionTTL,
-        packet: Arc<ImmutableDeserializedPacket>,
+        transaction_ttl: SanitizedTransactionTTL<Tx>,
+        packet: Option<Arc<ImmutableDeserializedPacket>>,
```

Comment:
plan is to NOT support forwarding w/ new transaction type, since forwarding in banking is short-lived

Reply:
Can we make this optional in a follow up change when it actually needs to be optional? Looks like it's always `Some` right now still